internal/pkg/vm/internal/vmcontext/syscalls.go

```diff
@@ -0,0 +1,92 @@
+package vmcontext
+
+import (
+	ffi "github.com/filecoin-project/filecoin-ffi"
```

Comment:
We shouldn't depend on ffi directly here, but an indirection so we can fake it for testing.

This interface already exists as verification.Verifier, hooked up to FFI and with a fake for testing. You can trace existing call paths into here to answer some of the questions about how to marshal parameters.